Source/interfaces/IDisplayInfo.h

```diff
@@ -33,6 +33,34 @@ namespace Exchange {
         virtual uint64_t FreeGpuRam() const = 0;
     };
 
+    struct EXTERNAL IDisplayProperties : virtual public Core::IUnknown {
+        enum { ID = ID_DISPLAY_PROPERTIES };
+
+        virtual ~IDisplayProperties() {}
+
+        struct EXTERNAL INotification : virtual public Core::IUnknown {
+            enum { ID = ID_DISPLAY_PROPERTIES_NOTIFICATION };
+
+            virtual ~INotification() {}
+
+            virtual void ResolutionChanged(const int32_t&, const int32_t&, const string&, const string& ) = 0;
+            virtual void ResolutionPreChange() = 0;
+        };
+
+        virtual uint32_t Register(INotification* sink) = 0;
+        virtual uint32_t Unregister(INotification* sink) = 0;
+
+        virtual string getCurrentResolution() const = 0;
+        virtual string getConnectedVideoDisplay() const = 0;
+        virtual void getTvHdrCapabilities(string& hdrcaps /* @inout */, int& caps /* @inout */) const = 0;
```

Comment:
The design/architectural principles are to create small granular blocks of functionality that can be reused for multiple customers and guide developpers during development.
HDMI functionality is part of the IConnectionProperties, so please put functionality for the connection on the IConnectionProperties.
As you can see the Iconnection properties has already some HDR properties. So I think the added methods here are actually part of the IConnection Properties.
From a development perspective, it is good to be specific. Why not use the enums for HDR capabilities ?
This methods are quite ambigous. Why should you send hdrcaps to it and receive hdrcaps from the implementation if you are getting ? I would expect here only /* @out */ ? Also if it is a list (that you are relaying over the interface) suggest iterators (See https://github.com/rdkcentral/Thunder/blob/master/Source/com/IRPCIterator.h using enums as a template paramter) allowing for strong typed feedback.

Reply:
> My intention behind adding all the new APIs under IDisplayProperties was in fact to keep the IConnectionProperties untouched. The reason being that if I add more functions to IConnectionProperties, it would instantly break the AmLogic, Rpi and Nexus implementations as these are derived from IConnectionProperties (unless stubs are written for them).

Left or right you need to add those properties to either the IDisplayProperties or to the IConnectionProperties for all the platforms. As we say in Holland: "Das lood om oud ijzer". So from an implementation perspective it is the same amount of work.

Reply:
> Surely, I could've done the same and added HLG to the enum list. But, that begged the question about how we keep this list exhaustive? For instance, tomorrow my device could return a new HDR format - say HDR20. In such a case I would need to add HDR20 as an enum inside Thunder before I can use it inside DisplayInfo plugin. Moreover, it would essentially result in constantly updating the HDR enum list within Thunder to keep it in sync with the enum list within DeviceSettings

In code you can only "act" on something if you know what it is. If a new standard is set forth, it means that new processing is required. The fact that the enum value for this future feature, and thus future processing is not available, is a good feature. If for example, somehwere you need to translate the HDR type/setting from the input to the enum and you can not convert it to that enum, I suggest rasing a serious log line there. If you do not prefer to log it there but in you plugin (I would agree to that) I would add a HDR catch all enum kuje HDR_UNKNOWN value. If you receive this value, you could raise an error n the plugin, like: Encountered an HDR mode for which I have no handling code. Than anyone reading the logs can already see why a HDR feature is not working. 
Also if it is a string, all code using this information   need to parse this string to determine what the should do with it.
As a rule of Thumb: Thunder development is DEFENSIVE. Which means use lots of ASSERTS where you know you are not catering for those scenarios and be TYPE SAFE. If possible use strong typed values (like enums) and not opaque objects like strings in which you could put in anything.
According to the V-Model of software development, the sooner you detect a bug (missing functionality) to sooner you raise it, the easier and cheaper it is to resolve it.

Reply:
And if I look at what we already have, it looks like the list is quite complete already:
https://github.com/rdkcentral/Thunder/blob/master/Source/interfaces/IDisplayInfo.h#L41-L47
Do like the addion of invalid, whihc I guess is the UNKNOWN.